hms-api/src/repository/hackathon-repository.ts

```diff
@@ -32,12 +34,39 @@ export async function listHackathons(): Promise<Hackathon[]> {
 }
 
 export async function putHackathon(hackathon: Hackathon) {
+  if (await hackathonExists(hackathon.id)) {
```

Comment:
I would say the same thing about this as I did about the IdeaComment that when you create a new Hackathon object the Uuid will be generated then and for sure (statistically very close to 0) won't match any existing ones so the check is a bit overkill.  And in the case where you are editing the hackathon needs to exist for you to modify it